File: Auto-Part/BackEnd/src/main/java/lk/ijse/autopart/rest/service/SellerAccountService.java
package lk.ijse.autopart.rest.service;

import lk.ijse.autopart.rest.dto.SellerAccountDTO;

import java.util.ArrayList;

public interface SellerAccountService {

    public ArrayList<SellerAccountDTO> getAllSellerAcccounts();

    public SellerAccountDTO getSellerAccounts(String aNic);

    public boolean deleteSellerAccount(String aNic);

    public boolean saveSellerAccount(SellerAccountDTO sellerAccount);

    public long getAllSellerAccount();
}

File: Auto-Part/FrontEnd/Admin/src/app/services/seller-account.service.ts
import { Injectable } from '@angular/core';
import {HttpClient} from "@angular/common/http";
import {Observable} from "rxjs";
import {Accounts} from "../dtos/accounts";

export const MAIN_URL ="http://localhost:8080";
const URL="/api/v1/sellerAccounts";

@Injectable()
export class SellerAccountService {

  constructor(private http:HttpClient) { }

  getAllSellerAcccounts():Observable<Array<Accounts>>{
    return this.http.get<Array<Accounts>>(MAIN_URL+URL);
  }

  deleteSellerAccount(aNic:string):Observable<boolean>{
    return this.http.delete<boolean>(MAIN_URL+URL+"/"+aNic);
  }

  saveSellerAccount(sellerAccount:Accounts):Observable<boolean>{
    return this.http.post<boolean>(MAIN_URL+URL,sellerAccount);
  }

  getSellerAccounts(aemail:string):Observable<Array<Accounts>>{
    return this.http.get<Array<Accounts>>(MAIN_URL+URL+"/"+aemail);
  }
  getAllSellerAccount():Observable<number>{
    return this.http.get<number>(MAIN_URL+URL+"/count");
  }

}
